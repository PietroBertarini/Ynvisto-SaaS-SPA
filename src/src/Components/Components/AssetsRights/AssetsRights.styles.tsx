import styled from 'styled-components';

export interface CardProps {
  color: string;
  children ?: any;
}

export const Main = styled.div<CardProps>`
  width: 100%;
  border-radius: 20px;
  margin-bottom: 2vh;
`;

export const MainDiv = styled.div`
  width: 100%;
  min-height: 100vh;
`;

export const Title = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  text-align: center;

  .values {
    display: flex;
    flex-direction: row;
    flex: 1;
    padding: 0 5%;
    justify-content: space-between;

    margin: 50px 0;
    color: #777;

    strong {
      color: #05E985;
      font-size: 43px;
    }
    p {
      margin: 0;
      padding: 0;
      font-size: 19px;
    }

    /* .totalItens { align-self: flex-start; }
    .valor2020 { align-self: center; }
    .valor2020 { align-self: flex-end; } */
  }
`;

export const List = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  .list {
    overflow-x: scroll;
    overflow-y: hidden;
    white-space: nowrap;
    max-width: 95%;

    &::-webkit-scrollbar {
      background-color: transparent;
      width: 2px;
    }

    &::-webkit-scrollbar-thumb {
      border-radius: 7px;
      background-color: rgba(100, 100, 120, 0.3);
    }

    .table {
      width: 100%;
      display: block;
      cursor: grab;
      border-radius: 20px;

      table {
        border-radius: 20px;
        font-size: 20px;
        min-width: 100%;
        border-collapse: collapse;
        table-layout: fixed;

        thead {
          padding-bottom: 0;
          margin-bottom: 0;
          table-layout: fixed;
          padding-left: 20px;

          th {
            min-width: 100px;
            display: inline-block;
            text-align: left;
            color: #999;
            font-weight: normal;
            font-size: 23px;
            padding-right: 20px;
          }

          .large {
            min-width: 250px;  
            width: 350px;
          }

          .largeRight {
            min-width: 250px;  
            text-align: right;
          }
        }

        tbody {
          tr {
            td {
              min-width: 100px;
              border-bottom: 1px solid #999;
              display: inline-block;
              overflow-x: hidden;

              color: #fff;
              font-size: 23px;
              padding: 20px 20px 5px 0;

              .empty {
                width: 70%;
                display: flex;
                float: left;
                content: '';
                height: 3px;
                background-color: #eee;
                display: block;
                margin: 10px auto;
              }

              button {
                color: #03dffc;
                font-weight: bold;
                font-size: 22px;
                padding-left: 10px;
                border:none;
                background-color: transparent;
              }
            }

            .large {
              width: 350px;
              min-width: 250px;             
            }

            .largeRight {
              min-width: 250px;  
              text-align: right;
            }
            
            &:hover {
              background: rgba(106, 187, 233, 0.11);
              td { background: transparent; }
            }
          }
        }
      }
    }
  }
`;
