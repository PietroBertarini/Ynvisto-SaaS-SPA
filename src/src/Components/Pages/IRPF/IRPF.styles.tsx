import { createStyles, makeStyles } from '@material-ui/core';
import styled from 'styled-components';

export const Title = styled.div`
  color: #fff;
  font-size: 2.8vw;
  font-family: 'Zurich';
  padding-left: 239px;
`;

export const Main = styled.div`
  flex-grow: 1;
  margin-left: 239px;
`;

export const useStyles = makeStyles(() => createStyles({
  header: {
    color: '#FFFFFF',
    fontSize: '2.8vw',
    fontFamily: 'Zurich',
    marginBottom: '2vh',
  },
}));
