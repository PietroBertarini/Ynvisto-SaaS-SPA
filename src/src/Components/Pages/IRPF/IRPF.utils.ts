export const allTabs = [
  'Bens e Direitos',
  'Rend. Isentos e Não Tributáveis',
  'Rend. Trib. Recebidos de PF/Exterior',
  'Rend. Suj. à Tributação Exclusiva/Definitiva',
  'Operações Comuns/Daytrade',
  'Operações FII',
];

export function getInitialValueTab(url: string | undefined): number {
  const parsed = url?.replace('%20', ' ');
  if (parsed === 'Rend. Isentos e Não Tributáveis') return 1;
  if (parsed === 'Rend. Trib. Recebidos de PF Exterior') return 2;
  if (parsed === 'Rend. Suj. à Tributação Exclusiva Definitiva') return 3;
  if (parsed === 'Operações Comuns Daytrade') return 4;
  if (parsed === 'Operações FII') return 5;
  return 0;
}
