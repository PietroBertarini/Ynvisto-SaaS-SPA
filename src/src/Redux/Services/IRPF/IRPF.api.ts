interface IAssetsRightsApiModel {
  ticker: string;
  code: number;
  location: string;
  situation20: number;
  situation21: number;
  discrimitation: string;
}

export interface IRPFApiModel {
  assetsRights: IAssetsRightsApiModel[],
}

export interface IAssetsRights {
  ticker: string;
  code: number;
  location: string;
  situation20: number;
  situation21: number;
  discrimitation: string;
}

/**
 * Get all IRPF from api.
 */
export function getAssetsRights(): IAssetsRights[] {
  return [
    {
      ticker: 'AMBR11',
      code: 9,
      location: '249 - Estados Unidos',
      situation20: 1864.03,
      situation21: 105102.53,
      discrimitation: 'Alguma descrição',
    },
    {
      ticker: 'ITAU5',
      code: 9,
      location: '105 - Brasil',
      situation20: 12359.95,
      situation21: 16120.53,
      discrimitation: 'Alguma descrição2',
    },
    {
      ticker: 'BBSE12',
      code: 9,
      location: '105 - Brasil',
      situation20: 3495.30,
      situation21: 4495.35,
      discrimitation: 'Alguma descrição3',
    },
  ];
}
