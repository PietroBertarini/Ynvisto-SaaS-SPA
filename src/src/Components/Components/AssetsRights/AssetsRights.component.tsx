import React, {
  useCallback, useState, memo, useEffect,
} from 'react';
import { Tooltip, withStyles } from '@material-ui/core';
import CheckCircleOutlineIcon from '@material-ui/icons/CheckCircleOutline';
import { IAssetsRights } from '../../../Redux/Services/IRPF/IRPF.api';
import {
  MainDiv, Title, List,
} from './AssetsRights.styles';

interface Props {
  assetsRights: IAssetsRights[],
}

const CustomTooltip = withStyles(() => ({
  tooltip: {
    backgroundColor: '#000',
    color: '#FFF',
    fontSize: 18,
  },
  arrow: {
    color: '#000',
  },
}))(Tooltip);

const AssetsRights = ({ assetsRights }: Props) => {
  let st20 = 0;
  let st21 = 0;

  if (assetsRights.length > 0) {
    assetsRights.forEach((item) => {
      st20 += item.situation20;
      st21 += item.situation21;
    });
  }

  const [situation20, setSituation20] = useState(st20);
  const [situation21, setSituation21] = useState(st21);

  useEffect(() => {
    setSituation20(st20);
    setSituation21(st21);
  }, [st20, st21]);

  let counter = 0;

  const copyToClipboard = useCallback((value: string) => {
    // Creates a dummy element, so we can copy it's value to clipboard
    const dummy = document.createElement('textarea');
    document.body.appendChild(dummy);
    dummy.value = value;
    dummy.select();
    document.execCommand('copy');
    document.body.removeChild(dummy);
  }, []);

  return (
    <MainDiv>
      <Title>
        <div className="values">
          <div className="totalItens">
            <strong>
              {assetsRights && (assetsRights.length < 10 ? `0${assetsRights.length}` : assetsRights.length)}
              {!assetsRights && '00'}
            </strong>
            <p>Total de Itens</p>
          </div>
          <div className="valor2020">
            <strong>
              {situation20.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
            </strong>
            <p>Situação em 31/12/2020</p>
          </div>
          <div className="valor2021">
            <strong>
              {situation21.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
            </strong>
            <p>Situação em 31/12/2021</p>
          </div>
        </div>
      </Title>
      {assetsRights.length !== 0 && (
      <List>
        <div className="list">
          <div className="table">
            <table>
              <thead>
                <tr>
                  <th style={{ height: 58 }}> </th>
                  <th>Item</th>
                  <th style={{ width: 150 }}>Ticker</th>
                  <th>Código</th>
                  <th className="large">Localização</th>
                  <th className="largeRight">Situação 2020</th>
                  <th className="largeRight">Situação 2021</th>
                  <th>Discriminação</th>
                </tr>
              </thead>
              <tbody>
                {assetsRights
                  && assetsRights.map((item) => {
                    counter += 1;
                    return (
                      <tr key={counter}>
                        <td style={{
                          textAlign: 'left',
                          margin: 0,
                          padding: 0,
                          height: 58,
                          paddingTop: 10,
                          paddingLeft: 10,
                          marginLeft: 10,
                        }}
                        >
                          <CheckCircleOutlineIcon
                            style={{
                              fontSize: 35, color: '#999', margin: 0, padding: 0,
                            }}
                          />
                        </td>

                        <td>{counter < 10 ? `0${counter}` : counter}</td>
                        <td style={{ width: 150 }}>
                          <CustomTooltip
                            title="Copiar"
                            arrow
                            placement="top-start"
                            onClick={() => copyToClipboard(item.ticker)}
                          >
                            <div className="text">
                              {item.ticker}
                            </div>
                          </CustomTooltip>
                        </td>
                        <td>
                          <CustomTooltip
                            title="Copiar"
                            arrow
                            placement="top-start"
                            onClick={() => {
                              if (item.code < 10) copyToClipboard(`0${item.code}`);
                              else copyToClipboard(item.code.toString());
                            }}
                          >
                            <div className="text">
                              {item.code < 10 ? `0${item.code}` : item.code}
                            </div>
                          </CustomTooltip>
                        </td>
                        <td className="large">
                          <CustomTooltip
                            title="Copiar"
                            arrow
                            placement="top-start"
                            onClick={() => copyToClipboard(item.location)}
                          >
                            <div className="text">
                              {item.location}
                            </div>
                          </CustomTooltip>
                        </td>
                        <td className="largeRight">
                          <CustomTooltip
                            title="Copiar"
                            arrow
                            placement="top-end"
                            onClick={() => copyToClipboard(item.situation20
                              .toLocaleString('pt-br', { style: 'currency', currency: 'BRL' }))}
                          >
                            <div className="text">
                              {item.situation20.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
                            </div>
                          </CustomTooltip>
                        </td>
                        <td className="largeRight">
                          <CustomTooltip
                            title="Copiar"
                            arrow
                            placement="top-end"
                            onClick={() => copyToClipboard(item.situation21
                              .toLocaleString('pt-br', { style: 'currency', currency: 'BRL' }))}
                          >
                            <div className="text">
                              {item.situation21.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
                            </div>
                          </CustomTooltip>
                        </td>
                        <td>
                          {item.discrimitation ? (
                            <button
                              type="button"
                              onClick={() => copyToClipboard(item.discrimitation)}
                            >
                              Copiar
                            </button>
                          ) : (
                            <div className="empty" />
                          )}
                        </td>
                      </tr>
                    );
                  })}
              </tbody>
            </table>
          </div>
        </div>
      </List>
      )}
    </MainDiv>
  );
};

export default memo(AssetsRights);
