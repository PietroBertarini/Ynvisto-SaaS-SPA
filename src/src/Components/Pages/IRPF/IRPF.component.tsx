import React from 'react';
import { Typography } from '@material-ui/core';
import { connect, ConnectedProps } from 'react-redux';
import { useParams } from 'react-router-dom';
import { State } from '../../../Redux/root-reducer';
import ScrollTabs from '../../Components/ScrollTabs/ScrollTabs.component';
import AssetsRights from '../../Components/AssetsRights/AssetsRights.component';
import { allTabs, getInitialValueTab } from './IRPF.utils';
import {
  Main, useStyles,
} from './IRPF.styles';

const mapStateToProps = (state: State) => ({
  assetsRights: state.irpf.assetsRights,
});

const connector = connect(mapStateToProps);
type PropsFromRedux = ConnectedProps<typeof connector>
type Props = PropsFromRedux

function IRPF(props: Props) {
  const { assetsRights } = props;

  const { page } = useParams<{ page: string }>();
  const navBarActive = getInitialValueTab(page);

  const classes = useStyles();

  return (
    <Main>
      <Typography className={classes.header}>
        IRPF - Imposto de Renda Pessoa FÃ­sica
      </Typography>
      <ScrollTabs initialTabValue={navBarActive} tabsLabel={allTabs} />
      {navBarActive === 0 && <AssetsRights assetsRights={assetsRights} />}
    </Main>
  );
}

export default connector(IRPF);
