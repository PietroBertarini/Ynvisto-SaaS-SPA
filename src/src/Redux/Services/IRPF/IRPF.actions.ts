import { IAssetsRights } from './IRPF.api';

export interface IRPF {
  assetsRights: IAssetsRights[],
}

export enum EIrpfTypes {
  GET_ASSETS_RIGHTS = 'GET_ASSETS_RIGHTS',
}

export interface IGetAssetsRights {
  type: typeof EIrpfTypes.GET_ASSETS_RIGHTS,
}

export type MyIRPFActionTypes = IGetAssetsRights;

export const getAssetsRightsAction = () : IGetAssetsRights => ({
  type: EIrpfTypes.GET_ASSETS_RIGHTS,
});
