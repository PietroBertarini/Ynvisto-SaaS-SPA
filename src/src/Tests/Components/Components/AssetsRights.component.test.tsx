import React from 'react';
import { shallow } from 'enzyme';
import AssetsRights from '../../../Components/Components/AssetsRights/AssetsRights.component';
import { mockStock } from '../../Redux/Components/Pages/IRPF/IRPF.mock';

test('Renders AssetsRights component and match to previous snapshot', () => {
  const component = shallow(<AssetsRights assetsRights={mockStock.assetsRights} />);
  expect(component).toMatchSnapshot();
});
