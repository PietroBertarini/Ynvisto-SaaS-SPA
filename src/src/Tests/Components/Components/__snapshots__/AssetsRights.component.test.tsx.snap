// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders AssetsRights component and match to previous snapshot 1`] = `
<styled.div>
  <styled.div>
    <div
      className="values"
    >
      <div
        className="totalItens"
      >
        <strong>
          05
        </strong>
        <p>
          Total de Itens
        </p>
      </div>
      <div
        className="valor2020"
      >
        <strong>
          R$ 26.335,23
        </strong>
        <p>
          Situação em 31/12/2020
        </p>
      </div>
      <div
        className="valor2021"
      >
        <strong>
          R$ 2.055,20
        </strong>
        <p>
          Situação em 31/12/2021
        </p>
      </div>
    </div>
  </styled.div>
  <styled.div>
    <div
      className="list"
    >
      <div
        className="table"
      >
        <table>
          <thead>
            <tr>
              <th
                style={
                  Object {
                    "height": 58,
                  }
                }
              >
                 
              </th>
              <th>
                Item
              </th>
              <th
                style={
                  Object {
                    "width": 150,
                  }
                }
              >
                Ticker
              </th>
              <th>
                Código
              </th>
              <th
                className="large"
              >
                Localização
              </th>
              <th
                className="largeRight"
              >
                Situação 2020
              </th>
              <th
                className="largeRight"
              >
                Situação 2021
              </th>
              <th>
                Discriminação
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              key="1"
            >
              <td
                style={
                  Object {
                    "height": 58,
                    "margin": 0,
                    "marginLeft": 10,
                    "padding": 0,
                    "paddingLeft": 10,
                    "paddingTop": 10,
                    "textAlign": "left",
                  }
                }
              >
                <CheckCircleOutlineIcon
                  style={
                    Object {
                      "color": "#999",
                      "fontSize": 35,
                      "margin": 0,
                      "padding": 0,
                    }
                  }
                />
              </td>
              <td>
                01
              </td>
              <td
                style={
                  Object {
                    "width": 150,
                  }
                }
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    AMBR11
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td>
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    09
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="large"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    249 - Estados Unidos da América
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="largeRight"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-end"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    R$ 29,03
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="largeRight"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-end"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    R$ 102,53
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td>
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Copiar
                </button>
              </td>
            </tr>
            <tr
              key="2"
            >
              <td
                style={
                  Object {
                    "height": 58,
                    "margin": 0,
                    "marginLeft": 10,
                    "padding": 0,
                    "paddingLeft": 10,
                    "paddingTop": 10,
                    "textAlign": "left",
                  }
                }
              >
                <CheckCircleOutlineIcon
                  style={
                    Object {
                      "color": "#999",
                      "fontSize": 35,
                      "margin": 0,
                      "padding": 0,
                    }
                  }
                />
              </td>
              <td>
                02
              </td>
              <td
                style={
                  Object {
                    "width": 150,
                  }
                }
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    ITAU5
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td>
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    09
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="large"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    105 - Brasil
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="largeRight"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-end"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    R$ 8.574,95
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="largeRight"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-end"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    R$ 132,53
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td>
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Copiar
                </button>
              </td>
            </tr>
            <tr
              key="3"
            >
              <td
                style={
                  Object {
                    "height": 58,
                    "margin": 0,
                    "marginLeft": 10,
                    "padding": 0,
                    "paddingLeft": 10,
                    "paddingTop": 10,
                    "textAlign": "left",
                  }
                }
              >
                <CheckCircleOutlineIcon
                  style={
                    Object {
                      "color": "#999",
                      "fontSize": 35,
                      "margin": 0,
                      "padding": 0,
                    }
                  }
                />
              </td>
              <td>
                03
              </td>
              <td
                style={
                  Object {
                    "width": 150,
                  }
                }
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    BBSE12
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td>
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    09
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="large"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    105 - Alguma frase bem longa
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="largeRight"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-end"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    R$ 6.397,30
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="largeRight"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-end"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    R$ 729,35
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td>
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Copiar
                </button>
              </td>
            </tr>
            <tr
              key="4"
            >
              <td
                style={
                  Object {
                    "height": 58,
                    "margin": 0,
                    "marginLeft": 10,
                    "padding": 0,
                    "paddingLeft": 10,
                    "paddingTop": 10,
                    "textAlign": "left",
                  }
                }
              >
                <CheckCircleOutlineIcon
                  style={
                    Object {
                      "color": "#999",
                      "fontSize": 35,
                      "margin": 0,
                      "padding": 0,
                    }
                  }
                />
              </td>
              <td>
                04
              </td>
              <td
                style={
                  Object {
                    "width": 150,
                  }
                }
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    OIES11
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td>
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    09
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="large"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    249 - Europa Central
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="largeRight"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-end"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    R$ 4.936,30
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="largeRight"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-end"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    R$ 134,35
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td>
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Copiar
                </button>
              </td>
            </tr>
            <tr
              key="5"
            >
              <td
                style={
                  Object {
                    "height": 58,
                    "margin": 0,
                    "marginLeft": 10,
                    "padding": 0,
                    "paddingLeft": 10,
                    "paddingTop": 10,
                    "textAlign": "left",
                  }
                }
              >
                <CheckCircleOutlineIcon
                  style={
                    Object {
                      "color": "#999",
                      "fontSize": 35,
                      "margin": 0,
                      "padding": 0,
                    }
                  }
                />
              </td>
              <td>
                05
              </td>
              <td
                style={
                  Object {
                    "width": 150,
                  }
                }
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    IXE8
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td>
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    09
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="large"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-start"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    105 - Brasil
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="largeRight"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-end"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    R$ 6.397,65
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td
                className="largeRight"
              >
                <WithStyles(WithStyles(ForwardRef(Tooltip)))
                  arrow={true}
                  onClick={[Function]}
                  placement="top-end"
                  title="Copiar"
                >
                  <div
                    className="text"
                  >
                    R$ 956,44
                  </div>
                </WithStyles(WithStyles(ForwardRef(Tooltip)))>
              </td>
              <td>
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Copiar
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </styled.div>
</styled.div>
`;
